sdk/core/Azure.Core/src/Shared/EventSourceEventFormatting.cs

```diff
@@ -56,5 +71,21 @@ private static object FormatValue(object o)
 
             return o;
         }
+
+        [EventSource(Name = EventSourceName)]
+        private class EventSourceEventFormattingEventSource : EventSource
```

Comment:
This won't work because the EventSourceEventFormatting.cs is included into multiple assemblies as a source and would result in multiple event sources with the same name.

Let's go back to the solution that you had initially with just suppressing the error and logging message format.

We can think about how to easier surface bugs in our code later.